src/m365/spo/commands/folder/folder-add.ts

```diff
@@ -86,39 +91,84 @@ class SpoFolderAddCommand extends SpoCommand {
       const serverRelativeUrl: string = `${parentFolderServerRelativeUrl}/${args.options.name}`;
 
-      const requestOptions: CliRequestOptions = {
-        headers: {
-          'accept': 'application/json;odata=nometadata'
-        },
-        responseType: 'json'
-      };
-
-      if (args.options.color === undefined) {
-        requestOptions.url = `${args.options.webUrl}/_api/web/folders/addUsingPath(decodedUrl='${formatting.encodeQueryParameter(serverRelativeUrl)}')`;
-      }
-      else {
-        requestOptions.url = `${args.options.webUrl}/_api/foldercoloring/createfolder(DecodedUrl='${formatting.encodeQueryParameter(serverRelativeUrl)}', overwrite=false)`;
-        requestOptions.data = {
-          coloringInformation: {
-            ColorHex: FolderColorValues[args.options.color] || args.options.color
-          }
-        };
+      if (args.options.ensureParentFolders) {
+        await this.ensureParentFolderPath(args.options, parentFolderServerRelativeUrl, logger);
       }
 
-      const folder = await request.post<FolderProperties>(requestOptions);
+      const folder = await this.addFolder(serverRelativeUrl, args.options, logger);
       await logger.log(folder);
     }
     catch (err: any) {
       this.handleRejectedODataJsonPromise(err);
     }
   }
+
+  private async ensureParentFolderPath(options: Options, parentFolderPath: string, logger: Logger): Promise<void> {
+    if (this.verbose) {
+      await logger.logToStderr(`Ensuring parent folders exist...`);
+    }
+
+    const parsedUrl = new URL(options.webUrl);
+    const absoluteFolderUrl: string = `${parsedUrl.origin}${parentFolderPath}`;
+    const relativeFolderPath = absoluteFolderUrl.replace(options.webUrl, '');
+
+    const parentFolders: string[] = relativeFolderPath.split('/').filter(folder => folder !== '');
+
+    for (let i = 1; i < parentFolders.length; i++) {
+      const currentFolderPath = parentFolders.slice(0, i + 1).join('/');
+
+      const folderExists = await this.getFolderExists(options.webUrl, currentFolderPath);
+      if (!folderExists) {
+        await this.addFolder(currentFolderPath, options, logger);
+      }
+    }
+  }
+
+  private async getFolderExists(webUrl: string, folderServerRelativeUrl: string): Promise<boolean> {
+    const requestUrl = `${webUrl}/_api/web/GetFolderByServerRelativePath(DecodedUrl='${formatting.encodeQueryParameter(folderServerRelativeUrl)}')?$select=Exists`;
+
+    const requestOptions: CliRequestOptions = {
+      url: requestUrl,
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    const response = await request.get<{ Exists: boolean }>(requestOptions);
+    return response.Exists;
+  }
+
+  private async addFolder(serverRelativeUrl: string, options: Options, logger: Logger): Promise<FolderProperties> {
+    if (this.verbose) {
+      await logger.logToStderr(`Adding folder to site ${options.webUrl}...`);
```

Comment:
Currently, this doesn't say a lot. Wouldn't it be better to mention the name of the folder here and the location of the folder?
Example:
Adding folder with name {name} at location {location}...

This way, we will also get a pretty tree structure if we want to ensure the parent folders, example for a structure A/B/C/D: 

Adding folder with name B at location A...
Adding folder with name C at location A/B...
Adding folder with name D at location A/B/C...

This says a lot more than three times a log that says Adding folder to site {webUrl}...